src/malli/core.cljc

```diff
@@ -174,7 +174,8 @@
                                (fn [[key {:keys [optional]} value]]
                                  (let [valid? (-validator value)
                                        default (boolean optional)]
-                                   (fn [m] (if-let [v (key m)] (valid? v) default))))
+                                   (fn [m] (let [v (key m ::missing-key)]
+                                             (if (not= v ::missing-key) (valid? v) default)))))
```

Comment:
Good catch! Equality for Keywords is slow on Clojure, I recall `identical?` is faster. Still, here we could use `find` to get the whole map entry, it's `nil` or `Map$Entry` and you can get the value effectively with `val`.

Reply:
If you could change that, would be happy merge this in.

Reply:
I've used `find` as that keeps the logic simpler and closer to the original.

Reply:
Let me know if you'd like further changes.